VMAccess/vmaccess.py

```diff
@@ -397,29 +399,26 @@ def _fsck_check(hutil):
         hutil.do_exit(1, 'Check', 'error', '0', 'Check failed.')
 
 
-def _fsck_repair(hutil):
+def _fsck_repair(disk_name, hutil):
     # first unmount disks and loop devices lazy + forced
     try:
-        cmd_result = waagent.Run("umount -adfr")
+        cmd_result = waagent.Run("umount -f /%s" % disk_name)
         if cmd_result!=0:
             # Fail fast
-            raise Exception("Failed to unmount disks")
+            hutil.log("Failed to unmount disk")
```

Comment:
I think, this line maybe needs to be removed because the error is logged in line 420.